### PR TITLE
Remove GeometryCreators, and use Geometries in place of them

### PR DESCRIPTION
**Pull request by biotinker**
Also remove `Verticies` from the Geometry interface.

One of those lovely PRs that deletes more lines than it adds.

### CODE REVIEW

File: spatialmath/sphere.go [hunk 1]
```diff
@@ -14,68 +14,34 @@ import (
...
 	pose   Pose
 	radius float64
 	label  string
 }
 
-// NewSphereCreator instantiates a SphereCreator class, which allows instantiating spheres given only a pose which is applied
-// at the specified offset from the pose. These spheres have a radius specified by the radius argument.
-func NewSphereCreator(radius float64, offset Pose, label string) (GeometryCreator, error) {
+// NewSphere instantiates a new sphere Geometry.
+func NewSphere(radius float64, offset Pose, label string) (Geometry, error) {
```
Comment (spieswl) [inline spatialmath/sphere.go:25]: Suggest swapping offset and radius ordering so that Boxes and Spheres both have their Pose arguments first.

**Review by spieswl — CHANGES_REQUESTED** [review]
Just a minor change to align argument order for the updated methods

File: spatialmath/sphere.go (hunk 1 above)
Comment (biotinker) [reply spatialmath/sphere.go:25]: Good catch, definitely should do that.

**Review by spieswl — APPROVED** [review]
LGTM!

**Pull request merged**